packages/react-i18n/src/i18n.ts

```diff
@@ -490,6 +499,48 @@ export class I18n {
     }
   }
 
+  /**
+   * Format a Date object to a human readable string, containing both date and time.
+   * It uses the `date.humanize.datetime` format that should be defined in the translations.
+   *
+   * Polaris describes the format that should be used for date/time combinations here:
+   * https://polaris.shopify.com/foundations/content/grammar-and-mechanics#date
+   *
+   * @param {Date} date Date object to format
+   * @param {Intl.DateTimeFormatOptions} [options={}] Options for both date and time formatting
+   * @return {*}  {string} A formatted string
+   * @memberof I18n
+   */
+  private formatDateTime(
+    date: Date,
+    options: Intl.DateTimeFormatOptions,
+  ): string {
+    const {defaultTimezone} = this;
+    const {timeZone = defaultTimezone} = options;
+
+    const {hour, minute, second, hour12, ...dateOptions} = options;
+    const {
+      weekday,
+      era,
+      year,
+      month,
+      day,
+      timeZoneName,
+      ...timeOptions
+    } = options;
+
+    return this.translate('date.humanize.datetime', {
```

Comment:
I suggest possibly reusing `date.humanize.lessThanOneYearAgo` - for existing users that already have this key translated, it will mean not having to add another key.

Reply:
👍 Agreed!